pkg/plugin/work-in-progress/event_handler.go

```diff
@@ -146,35 +119,69 @@ func (gh *GitHubWIPPRHandler) handlePrComment(log log.Logger, comment *gogh.Issu
 	return err
 }
 
-func (gh *GitHubWIPPRHandler) checkTitleAndSetStatus(log log.Logger, pullRequest *gogh.PullRequest) error {
-	change := scm.RepositoryChange{
-		Owner:    *pullRequest.Base.Repo.Owner.Login,
-		RepoName: *pullRequest.Base.Repo.Name,
-		Hash:     *pullRequest.Head.SHA,
-	}
+func (gh *GitHubWIPPRHandler) checkComponentsAndSetStatus(log log.Logger, pullRequest *gogh.PullRequest, labelUpdated bool) error {
+	change := ghservice.NewRepositoryChangeForPR(pullRequest)
 	statusContext := github.StatusContext{BotName: gh.BotName, PluginName: ProwPluginName}
 	statusService := ghservice.NewStatusService(gh.Client, log, change, statusContext)
 
-	labels, err := gh.Client.ListPullRequestLabels(change, *pullRequest.Number)
-	if err != nil {
-		log.Warnf("failed to list labels on PR [%q]. cause: %s", *pullRequest, err)
-	}
-
 	configuration := LoadConfiguration(log, change)
-	labelExists := gh.hasWorkInProgressLabel(labels, configuration.Label)
-	if gh.IsWorkInProgress(*pullRequest.Title, configuration) {
-		if !labelExists {
-			if err := gh.Client.AddPullRequestLabel(change, *pullRequest.Number, strings.Fields(configuration.Label)); err != nil {
-				log.Errorf("failed to add label on PR [%q]. cause: %s", *pullRequest, err)
+	labelExists := gh.hasWorkInProgressLabel(pullRequest.Labels, configuration.Label)
+	prefixExists, prefix := gh.HasWorkInProgressPrefix(*pullRequest.Title, configuration)
+
+	if prefixExists && !labelExists {
+		if labelUpdated {
+			*pullRequest.Title = strings.TrimPrefix(*pullRequest.Title, prefix)
+			if err := gh.Client.EditPullRequest(pullRequest); err != nil {
+				return fmt.Errorf("failed to update PR title [%q]. cause: %s", *pullRequest, err)
 			}
+			return statusService.Success(ReadyForReviewMessage, ReadyForReviewDetailsPageName)
+		}
+		if err := gh.Client.AddPullRequestLabel(change, *pullRequest.Number, strings.Fields(configuration.Label)); err != nil {
```

Comment:
why `strings.Fields`? the label should be only one and can also contain a white space - eg: `work in progress`

Reply:
The following function used to add label expects a string array of labels. We only have one string label and that needs to be converted to a string array and hence the choice of `string.Field()`. 

The choice of converting the string to string array is done at this location instead inside the  `gh.Client.AddPullRequestLabel` because I wanted to keep this method unmodified considering this could be used elsewhere where we could have to send multiple label values. (Example: https://github.com/arquillian/ike-prow-plugins/issues/57) 
  
```
// AddLabelsToIssue adds labels to an issue.
//
// GitHub API docs: https://developer.github.com/v3/issues/labels/#add-labels-to-an-issue
func (s *IssuesService) AddLabelsToIssue(ctx context.Context, owner string, repo string, number int, labels []string) ([]*Label, *Response, error) {
```

Reply:
Thanks for the explanation @MatousJobanek, updated it.